contracts/UmaConditionalTokensBinaryAdapter.sol

```diff
@@ -461,18 +413,62 @@ contract UmaCtfAdapter is Auth, BulletinBoard, ReentrancyGuard {
         emit QuestionReset(questionID);
     }
 
-    /// @notice Gets resolution data for the question from the OO
-    /// @param questionData - The parameters of the question
-    function _getResolutionData(QuestionData storage questionData) internal view returns (int256) {
+    /// @notice Resolves the underlying CTF market
+    /// @param questionID   - The unique questionID of the question
+    /// @param questionData - The question data parameters
+    function _resolve(bytes32 questionID, QuestionData storage questionData) internal {
+        // Get the price from the OO
+        int256 price = optimisticOracle.settleAndGetPrice(
```

Comment:
can someone settle the market in front of us? does that cause any issues with us calling this function?

Reply:
Yes but that will not cause any issues for us. If a request is already settled, settleAndGetPrice will just return the resolvedPrice